test/Dialect/QUIR/Transforms/merge-circuit-measures.mlir

```diff
@@ -0,0 +1,192 @@
+    %res5_0 = quir.call_circuit @circuit_0(%q0) : (!quir.qubit<1>) -> (i1)
+    %cast5_0 = "oq3.cast"(%res5_0) : (i1) -> !quir.cbit<1>
+    oq3.variable_assign @c0 : !quir.cbit<1> = %cast5_0
+    %res5_1 = quir.call_circuit @circuit_1(%q1) : (!quir.qubit<1>) -> (i1)
+    %cast5_1 = "oq3.cast"(%res5_1) : (i1) -> !quir.cbit<1>
+    oq3.variable_assign @c0 : !quir.cbit<1> = %cast5_1
+    %res5_2 = quir.call_circuit @circuit_2(%q2) : (!quir.qubit<1>) -> (i1)
+    %cast5_2 = "oq3.cast"(%res5_2) : (i1) -> !quir.cbit<1>
+    oq3.variable_assign @c0 : !quir.cbit<1> = %cast5_2
+    %res3 = quir.call_circuit @circuit_3(%q3) : (!quir.qubit<1>) -> (i1)
+
+    // CHECK:  quir.barrier %0, %1, %2, %3, %4, %5
+    quir.barrier %q0, %q1, %q2, %q3, %q4, %q5: (!quir.qubit<1>, !quir.qubit<1>, !quir.qubit<1>, !quir.qubit<1>, !quir.qubit<1>, !quir.qubit<1>) -> ()
+
+    // four_interrupted
+    // CHECK:  %{{.*}}:4 = quir.call_circuit @"circuit_0_q0_circuit_1_q1_circuit_2_q2_circuit_3_q3+m+m+m"(%{{.*}}, %{{.*}}, %{{.*}}, %{{.*}}) : (!quir.qubit<1>, !quir.qubit<1>, !quir.qubit<1>, !quir.qubit<1>) -> (i1, i1, i1, i1)
+    %res4_0 = quir.call_circuit @circuit_0(%q0) : (!quir.qubit<1>) -> (i1)
+    %cast4_0 = "oq3.cast"(%res4_0) : (i1) -> !quir.cbit<1>
+    oq3.variable_assign @c0 : !quir.cbit<1> = %cast4_0
+    %res4_1 = quir.call_circuit @circuit_1(%q1) : (!quir.qubit<1>) -> (i1)
+    %cast4_1 = "oq3.cast"(%res4_1) : (i1) -> !quir.cbit<1>
+    oq3.variable_assign @c0 : !quir.cbit<1> = %cast4_1
+
+    // CHECK: quir.call_gate @x(%{{.*}}) : (!quir.qubit<1>) -> ()
+    quir.call_gate @x(%q4) : (!quir.qubit<1>) -> ()
+
+    // CHECK-NOT:  %{{.*}} = quir.call_circuit @circuit_2
+    %res4_2 = quir.call_circuit @circuit_2(%q2) : (!quir.qubit<1>) -> (i1)
+    %cast4_2 = "oq3.cast"(%res4_2) : (i1) -> !quir.cbit<1>
+    oq3.variable_assign @c0 : !quir.cbit<1> = %cast4_2
+    %res4_3 = quir.call_circuit @circuit_3(%q3) : (!quir.qubit<1>) -> (i1)
+
+    // CHECK:  quir.barrier %0, %1, %2, %3, %4, %5
+    quir.barrier %q0, %q1, %q2, %q3, %q4, %q5: (!quir.qubit<1>, !quir.qubit<1>, !quir.qubit<1>, !quir.qubit<1>, !quir.qubit<1>, !quir.qubit<1>) -> ()
+
+    // inter_if
+    // CHECK: %{{.*}} = quir.call_circuit @circuit_0_q0(%0)
+    %res3_0 =  quir.call_circuit @circuit_0(%q0) : (!quir.qubit<1>) -> i1
+
+    %cond = arith.constant 1 : i1
+
+    scf.if %cond {
+      quir.call_circuit @circuit_cx(%q2, %q3) : (!quir.qubit<1>, !quir.qubit<1>) -> ()
+    }
+
+    // CHECK: %{{.*}}:2 = quir.call_circuit @"circuit_1_q1_circuit_2_q2+m0"(%{{.*}}, %{{.*}}) : (!quir.qubit<1>, !quir.qubit<1>) -> (i1, i1)
+    %res3_1 = quir.call_circuit @circuit_1(%q1) : (!quir.qubit<1>) -> (i1)
+    %res3_2 = quir.call_circuit @circuit_2(%q2) : (!quir.qubit<1>) -> (i1)
+
+    // CHECK:  quir.barrier %0, %1, %2, %3, %4, %5
+    quir.barrier %q0, %q1, %q2, %q3, %q4, %q5: (!quir.qubit<1>, !quir.qubit<1>, !quir.qubit<1>, !quir.qubit<1>, !quir.qubit<1>, !quir.qubit<1>) -> ()
```

Comment:
Using `quir.barriers` like this is my current method for isolating test cases so that the passes will not attempt to merge circuit measures past these barriers.